OrangeHRMTestFramework/Tests/OrangeHRM/UserManagementTests.cs

```diff
@@ -0,0 +1,217 @@
+
+            // Click Add user button to open Add User Page
+            var addUserPage = GenericPages.UserManagementPage.ClickAddUserButton();
+
+            // Enter the same data as for the first user
+            addUserPage.AddUserTab.EnterAndSelectValueInEmployeeNameFilterInput(_employeeName);
+            addUserPage.AddUserTab.EnterValueToPasswordAndConfirmPasswordInputs(userPassword);
+            addUserPage.AddUserTab.EnterDataToUsernameInput(userName);
+            addUserPage.AddUserTab.SelectValueInStatusDropdown(Status.Enabled);
+            addUserPage.AddUserTab.SelectValueInUserRoleDropdown(UserRole.Admin);
+            addUserPage.AddUserTab.ClickSaveButton();
+
+            // Check that warning message is displayed
+            var isWarningDisplayed = addUserPage.AddUserTab.IsWarningMessageWithCertainTextIsDisplayed(OrangeMessages.AlreadyExistsWarningMessage);
+            Assert.IsTrue(isWarningDisplayed);
+        }
+
+        [Test]
+        public void SortUserListByUserNameDescTest()
+        {
+            // Get cell values from the DataGrid before sorting
+            var userNamesListBeforeSorting = GenericPages.UserManagementPage.BasicDataGrid.GetCellValuesFromAllPages(UserManagementFieldNames.Username);
+
+            // Get expected result by  sorting list of values desc
+            var expectedResultList = userNamesListBeforeSorting.OrderByDescending(x => x);
+
+            // Sort Usernames on the Data Grid by Username desc
+            GenericPages.UserManagementPage.BasicDataGrid.SortDescByColumnName(UserManagementFieldNames.Username);
+
+            // Get cell values from the DataGrid after sorting
+            var userNamesListAfterSorting = GenericPages.UserManagementPage.BasicDataGrid.GetCellValuesFromAllPages(UserManagementFieldNames.Username);
+
+            // Check that values in the Grid are really sorted by desc
+            Assert.AreEqual(expectedResultList, userNamesListAfterSorting);
+        }
+
+        private string AddTestEmployeeAndGetName()
+        {
+            var firstName = RandomHelper.GetRandomString(7);
+            var lastName = RandomHelper.GetRandomString(7);
+            var middleName = RandomHelper.GetRandomString(7);
+            GenericPages.EmployeeManagementPage.ClickLeftNavCategory(LeftNavCategories.Pim);
+            GenericPages.EmployeeManagementPage.ClickTopNavCategoryWithoutSubCategory(PimTopNavCategories.EmployeeList);
+            var addEmployeePage = GenericPages.EmployeeManagementPage.ClickAddEmployeeButton();
+            addEmployeePage.AddEmployeeTab.EnterDataToAllInputsAndClickSave(firstName, lastName, middleName);
+            addEmployeePage.WaitUntilSuccessMessageDisplayed();
+
+            return $"{firstName} {lastName}";
+        }
+
+        private void AddTestUser(string userName, string userPassword, UserRole valueRole, Status valueStatus)
```

Comment:
Last 2 parameters are unused. You can remove them or make them default

Reply:
Sorry, fixed 🥲